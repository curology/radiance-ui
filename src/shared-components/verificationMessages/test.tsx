import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { VerificationMessages } from './index';

describe('<VerificationMessages />', () => {
  describe('UI snapshot', () => {
    it('renders with default props and some messages', () => {
      const messages = {
        required: (
          <React.Fragment>
            <strong>Uh oh!</strong> This field is required
          </React.Fragment>
        ),
        maxLength: (
          <React.Fragment>
            <strong>Uh oh!</strong> Must be at least 3 characters
          </React.Fragment>
        ),
        arrayString: ['Must contain 1 number', 'Must contain 1 symbol'],
        arrayJSX: [
          <React.Fragment key="1">
            <strong>1</strong> one
          </React.Fragment>,
          <React.Fragment key="2">
            <strong>2</strong> two
          </React.Fragment>,
        ],
      };

      const { container } = render(
        <VerificationMessages messages={messages} />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders with non-default props', () => {
      const messages = {
        required: (
          <React.Fragment>
            <strong>Congrats!</strong> Your app was approved
          </React.Fragment>
        ),
      };

      const { container } = render(
        <VerificationMessages messages={messages} centered type="success" />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('when message value is an empty array', () => {
    it('renders nothing', () => {
      const messages = {
        required: [],
      };

      const { queryAllByRole } = render(
        <VerificationMessages messages={messages} />,
      );

      expect(queryAllByRole('list')).toHaveLength(1);
      expect(queryAllByRole('listitem')).toHaveLength(0);
    });
  });
});
