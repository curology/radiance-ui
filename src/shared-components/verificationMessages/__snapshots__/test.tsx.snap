// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerificationMessages /> UI snapshot renders with default props and some messages 1`] = `
.emotion-0 {
  list-style-type: none;
  margin: 0;
}

.emotion-2 {
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  color: #BD200F;
  line-height: 24px;
}

.emotion-2:last-of-type {
  margin: 0 0 0.125rem 0;
}

<ul
  class="emotion-0 emotion-1"
>
  <li
    class="emotion-2 emotion-3"
    style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
    type="error"
  >
    <strong>
      Uh oh!
    </strong>
     This field is required
  </li>
  <li
    class="emotion-2 emotion-3"
    style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
    type="error"
  >
    <strong>
      Uh oh!
    </strong>
     Must be at least 3 characters
  </li>
  <li
    class="emotion-2 emotion-3"
    style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
    type="error"
  >
    Must contain 1 number
    , 
    Must contain 1 symbol
  </li>
  <li
    class="emotion-2 emotion-3"
    style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
    type="error"
  >
    <strong>
      1
    </strong>
     one
    , 
    <strong>
      2
    </strong>
     two
  </li>
</ul>
`;

exports[`<VerificationMessages /> UI snapshot renders with non-default props 1`] = `
.emotion-0 {
  list-style-type: none;
  margin: 0;
  text-align: center;
}

.emotion-2 {
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  color: #2B6E33;
  line-height: 24px;
}

.emotion-2:last-of-type {
  margin: 0 0 0.125rem 0;
}

<ul
  class="emotion-0 emotion-1"
>
  <li
    class="emotion-2 emotion-3"
    style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
    type="success"
  >
    <strong>
      Congrats!
    </strong>
     Your app was approved
  </li>
</ul>
`;
